Added requirements file (Should contain tentative essentials). Also organizing tif reader.

HGeerlings opened the pull request.

HGeerlings committed:
Added requirements. Also organizing tif reader

HGeerlings committed:
Added transpose for 3D arrays in tif-read

bkappes commented:
This wraps TIF-specific readers into `read.py`. We talked about `read.py` being an entry point that would attempt to guess the file format, and call the appropriate reader from `readers/tif.py`, `readers/dcm.py`, etc., didn't we?

bkappes commented:
Never mind! I missed the tif subdirectory.

bkappes commented on tracr/io/tif/read.py:
```diff
@@ -1,23 +1,51 @@
+        - Check if argument is single file (multi or single layer) or directory
+        - Call appropriate reader
+		- Tranpose data for upwards-z indexing
+    """
```
I see two that you've coded two use cases:

1. if a directory, then assume all files in that directory are lexicographically ordered TIF files and convert. 

2. If a filename: check whether the file is a single frame or a multilayer TIF.

How about a third case: 

3. List of files to process. This pushed the onus of `glob`bing the files onto the user, but provides a way for the user to more explicitly provide the image sequence.

It would be good to include a bit more of this level of detail in the doc string.

bkappes reviewed (changes requested):
As we discussed earlier today, I think it would be good to change the call signature of `read` in the `io` module to `def read(obj, **kwds)` where `obj` can be a single filename (`isinstance(obj, str)` + not directory), or a list of files (`isinstance(obj, (list, tuple, np.ndarray))`, or, if `obj` is a directory, creates a list of filenames from the contents of the directory. Then `read` in `tif/read.py` can expect either a single filename or a list of filenames.

HGeerlings committed:
Cleaned up some docs

HGeerlings committed:
Add command line usage to tif reader

HGeerlings committed:
Add bracket for ofile name

HGeerlings committed:
Ensure single tif frames are converted to greyscale before converting to array

HGeerlings committed:
small bug in greyscale conversion

HGeerlings committed:
Only greyscale RGB images

Ensuring that only RGB/RGBA type TIFs are greyscaled. Otherwise,
greyscaling of I type (tomographies) is damaging.

HGeerlings committed:
Initiate Feature class when returning arrays

HGeerlings committed:
Add pixelsize keyword for passing to Feature initialization

HGeerlings committed:
Only check 'tif' file listing rather than just listing all (glob(dir/*)) possible files

HGeerlings committed:
Add tif reading for iterable objects (if not folder, single/multi frame file). NExt is avoiding glob/listdir use for folders.

HGeerlings committed:
Update docstring. Replace folder/file reading with general list reading.

HGeerlings committed:
Move pixelsize check to tif specific reader. Value is set 1 if unspecified, and warning is logged.

HGeerlings committed:
Add warnings for any pixelsize. Ensure float pixelsize, break up array transpose.
